lib/config.js

```diff
@@ -723,6 +723,10 @@ util.loadFileConfigs = function() {
   return config;
 };
 
+function isDeferredConfig(value) {
+  return value instanceof DeferredConfig || (typeof value == 'object' && typeof value.resolve == 'function');
```

Comment:
A little bit stronger criterion would be better here, I think. What about `... || value.constructor.name === 'DeferredConfig'` ?